GitHub Action to lint Python code

cclauss opened the pull request:
Please annotate your PR title to describe what the PR does, then give a brief bulleted description of your PR below. PR titles should begin with [BUGFIX], [FEATURE],  [DOCS], or [MAINTENANCE]. If a new feature introduces breaking changes for the Great Expectations API or configuration files, please also add [BREAKING]. You can read about the tags in our [contributor checklist](https://docs.greatexpectations.io/en/latest/contributing/contribution_checklist.html).

Changes proposed in this pull request:
-
-
-


After submitting your PR, CI checks will run and @tiny-tim-bot will check for your CLA signature.

For a PR with nontrivial changes, we review with both design-centric and code-centric lenses.

In a design review, we aim to ensure that the PR is consistent with our relationship to the open source community, with our software architecture and abstractions, and with our users' needs and expectations. That review often starts well before a PR, for example in github issues or slack, so please link to relevant conversations in notes below to help reviewers understand and approve your PR more quickly (e.g. `closes #123`).

Previous Design Review notes:
-
-
-


### Definition of Done
Please delete options that are not relevant.

- [ ] My code follows the Great Expectations [style guide](https://docs.greatexpectations.io/en/latest/contributing/style_guide.html?highlight=style%20guide)
- [ ] I have performed a [self-review](https://docs.greatexpectations.io/en/latest/contributing/contribution_checklist.html?highlight=checklist) of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have added [unit tests](https://docs.greatexpectations.io/en/latest/contributing/testing.html#contributing-testing-writing-unit-tests) where applicable and made sure that new and existing tests are passing.
- [ ] I have run any local integration tests and made sure that nothing is broken.


Thank you for submitting!

cclauss committed:
GitHub Action to lint Python code

cdkini commented:
Hey @cclauss! Thanks for your contribution! 

Please note that the repository utilizes a pre-commit hook to perform linting: https://docs.greatexpectations.io/en/latest/contributing/style_guide.html#contributing-style-guide . In the case that pre-commit is somehow bypassed, the Azure CI/CD checks perform a lint check and block merging.

cclauss committed:
Fix an undefined name created in #2666

#2666 changed `df` into `data` but missed this instance which could lead to NameError being raised at runtime.

cclauss committed:
Fix an exception that is always True

cclauss committed:
Merge branch 'develop' into patch-1

cclauss committed:
Ignore undefined names

cclauss committed:
Move #3063 to a separate PR

#3063

cclauss committed:
pip install -r requirements-dev.txt fails

cdkini commented:
@cclauss did you happen to see my last comment regarding our current pre-commit hook configuration? Do you believe the changes herein are different than what we've set up already?

I do believe them to be duplicative but I'm happy to start a dialog.

Commit:
Merge branch 'develop' into patch-1

cdkini commented:
@cclauss I'm going to close this due to its overlap with our current pre-commit lint setup. Thanks for all the help you've provided so far 🚀
